File: Present from lena/present_from_lena.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

void Present_from_lena()
{ int N;
  cin>>N;
  
  //UPPER HALF
  for(int i = 0;i <= N ;i++ )
  {
      for(int j = i ;j < N;j++ )
      cout<<" "<<" ";
      if (i == 0)
         cout<<0;
      else{
         for(int j = 0;j <= i;j++)
            
              cout<<j<<" ";
      } 
      
      int j = i - 1 ;
      while (j >= 0)
      { if(j == 0)
         cout<<j;
        else 
          cout<<j<<" ";
          j -= 1;
          
      }
      cout<<"\n";
      
      
  }
  // LOWER HALF
  for(int i = 0 ;i < N ;i++)
  {
      for(int j = 0;j <= i;j++)
         cout<<"  ";
      if(i == N-1)
      cout<<0;
      else
      {
          for(int j = 0 ;j < N - i;j++)
          cout<<j<<" ";
      }
      int j = N - i - 2;
      while(j>=0)
      {if(j == 0)
       cout<<0;
       else
          cout<<j<<" ";
          j -= 1;
      }
      cout<<"\n";
      
      
      
  }
  
  
  
  
  
  
}


int main() {
	int remained_test_cases = 1 ;
// 	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   Present_from_lena();
	    remained_test_cases--;
	}
	return 0;
}

File: Present from lena/present_from_lena.py
def Present_from_Lena():
    N = int(input())


    #UPPER HALF
    for i in range(N+1):
        for z in range(i,N):
            print(" ",end=" ")
        for j in range(i):
            print(j,end=" ")
        j = i
        while j >= 0:
            if j == 0:
                print(j)
            else:    
                print(j,end=" ")
            j -= 1



    #LOWER HALF
    for i in range(N):
        for j in range(i+1):
            print(" ",end = " ")
        if i == N-1:
            print(0)
        else:    
            for j in range(N-i):
                print(j,end=" ")
            j -= 1     
            while j >= 0:
                if j == 0:
                    print(j)
                else:    
                    print(j,end= " ")
                j -= 1 
            
            
            
    
    
    
remained_test_cases = 1 
# remained_test_cases = int(input())
while remained_test_cases > 0:
    (Present_from_Lena())
    remained_test_cases -= 1 